### PR TITLE
fix: swap lat and lng

### LINKED ISSUE #2587: 🐛 [Bug] Geolocation. lat and lng are swapped
When selecting a location in the user profile page, the latitude and the longitude are exchanged when saved in the database.

### Steps to reproduce the behavior
1. Go to your profile
2. Select Configuration
3. Type in some City or Region
4. Save
5. Check the lat and lng properties in the location node
grafik

### Expected behavior
The value of lat should be in lng and reverse.

### PR DESCRIPTION
**Pull request by Mogge**
### Issues
- fixes #2587

### Todo
- [x] Find a solution how to repair the wrong data in production. Best would be to have script that calls the data by id from the geomap api for each location and merges the received data correctly in the lng and lat fields.

### CODE REVIEW

**Comment by mattwr18** [comment]
before deploying this, we need to do a database manipulation where we swap the existing locations, as @Mogge said in our pairing session.

This script works... 
```
MATCH (location:Location) with location.lng as longitude, location.lat as latitude, location 
SET location.lng = latitude, location.lat = longitude 
RETURN location;
```

**Comment by Mogge** [comment]
```cypher
MATCH (l:Location) WHERE NOT(l.lat IS NULL)
WITH  l.lng AS longitude, l.lat AS latitude, l AS Location
SET Location.lat = longitude, Location.lng = latitude
```

**Comment by mattwr18** [comment]
Why do we have `Location` with no `lat` property @Mogge ??

File: backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        SET Location.lat = longitude, Location.lng = latitude
+    `)
+    await transaction.commit()
+    next()
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(error)
+    await transaction.rollback()
+    // eslint-disable-next-line no-console
+    console.log('rolled back')
```
Comment (roschaefer) [inline backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js]:
```suggestion
    console.log('rolled back')
    throw new Error(error)
```

File: backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { getDriver } from '../../db/neo4j'
+
+export const description =
+  'This migration swaps the value stored in Location.lat with the value of Location.lng.'
+
+export async function up(next) {
```
Comment (roschaefer) [inline backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js]:
```suggestion
const swap = (next) => {
```

File: backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    console.log(error)
+    await transaction.rollback()
+    // eslint-disable-next-line no-console
+    console.log('rolled back')
+  } finally {
+    session.close()
+  }
+}
+
+export async function down(next) {
```
Comment (roschaefer) [inline backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js]: Instead:
```js
export async function up(next) {
  swap(next)
}
export async function down(next) {
  swap(next)
}
```

File: backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js (hunk 1 above)
Comment (Mogge) [reply backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js]: Done

File: backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js (hunk 2 above)
Comment (Mogge) [reply backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js]: Done

File: backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js (hunk 3 above)
Comment (Mogge) [reply backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js]: Done

File: backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+values lat=10.0 and lng=53.55, which is close to the horn of Africa,
+but it is lat=53.55 and lng=10.0
+`
+
+const swap = async function(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+  try {
+    // Implement your migration here.
```
Comment (mattwr18) [inline backend/src/db/migrations/20200206190233-swap_latitude_with_longitude.js]:
```suggestion
    
```
I would remove this line, but I won't request changes...

**Pull request merged**